worlds/stardew_valley/regions.py

```diff
@@ -372,24 +355,30 @@ def create_regions(region_factory: RegionFactory, random: Random, world_options:
 
 def randomize_connections(random: Random, world_options: StardewOptions) -> Tuple[List[ConnectionData], Dict[str, str]]:
     connections_to_randomize = []
+    final_connections = mandatory_connections
```

Comment:
Just because you assign it to a different name, doesn't make it a different object. This is a reference to the same object, and when you will extend final_connections, it will also extend mandatory_connections.

You can create a copy with list(mandatory_connections), or to be even cleaner, you can do

```
final_connections = []
final_connections.extend(mandatory_connections)
```